gpAux/gpperfmon/src/gpmon/gpsmon.c

```diff
@@ -861,7 +861,7 @@ static void gx_gettcpcmd(SOCKET sock, short event, void* arg)
 	struct timeval tv;
 	tv.tv_sec = opt.terminate_timeout;
 	tv.tv_usec = 0;
-	if (event_add(&gx.tcp_event, &tv)) //reset timeout
+	if (event_add(&gx.tcp_event, opt.terminate_timeout ? &tv : 0)) //reset timeout
```

Comment:
I would rather replace the else-branch with `NULL`.

Reply:
It makes sense but I tried to follow the general trend in this module

Reply:
Fixed